### ISSUE TITLE
is the blocked by dependency label really necessary?

### ISSUE THREAD

**Issue by inf3rno** [inf3rno/e2e-client#17]
I am not sure whether I need the `status: blocked - dependency` label. Every tracking issue will be blocked by its sub-issues with this approach. And if so, every sub-issue will have higher priority than the parent issue, because it blocks the parent.

**Comment by inf3rno** [comment]
I guess the block should be something unusual to use a block label, and I might need to rethink these status labels. It is ok that review and help needed is red, because it is easy to find them this way, but I am not sure whether they should be under the block group, or whether they should be separate status labels.

**Comment by inf3rno** [comment]
For now I think I'll add a `do: track` or an `on: it's complicated` label. So I won't need to block every complex issue whenever I move a part to a sub-issue. 

The goal of the `status: blocked` here to signal others that they should check the issue and solve the problem if they can. So help / review / etc... The blocking dependency is something that is needed for the actual issue, but it is not a sub-issue. For example we need a bug fix to be able to implement a new feature and we are not necessary the bug reporter. 

Hmm I think I can solve this. If a `do: debug` can't be a sub-issue of a `do: enhance`, then the bug will always block the feature issue as a dependency. If so, then I won't need a `do: track`, I just need a new rule. What if we are talking about a `do: enhance` and another `do: enhance`. Well that can be interesting. If the dependency is small, then it is simply a sub-issue. If it is big enough (it has it's own sub-issues), then it can be a blocking dependency. Usually this kind of dependencies are expected and if so, we can do a milestone for each of them. This way we don't have to use the blocked label, just filter by milestone.

We might need to rethink milestones and priority too.

**Comment by inf3rno** [comment]
I think it is better to document the labels somewhere, so I start a new issue for that. This can be closed.
